custom/git-app-deployer/main.go

```diff
@@ -97,23 +125,87 @@ func CreateSockShop(path string, ns string) error {
 	return nil
 }
 
+//CreateApplication creates application of sock-shop
+func CreateApplication(appVars *APPVars, delay int, clientset *kubernetes.Clientset) error {
+
+	log.Infof("[Status]: FilePath for App Deployer is %v", appVars.filePath)
+
+	if err := CreateNamespace(clientset, appVars.namespace); err != nil {
+		log.Info("[Status]: Namespace already exist!")
+	}
+
+	if err := UpdateSockShop("/var/run/"+appVars.filePath, appVars.namespace, appVars.operation); err != nil {
+		log.Errorf("Failed to install sock-shop, err: %v", err)
+		return err
+	}
+
+	if err := CheckApplicationStatus(appVars.namespace, "app=sock-shop", appVars.timeout, 2, appVars.operation, clientset); err != nil {
+		log.Errorf("err: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+//DeleteApplication deletes application of sock-shop
+func DeleteApplication(appVars *APPVars, delay int, clientset *kubernetes.Clientset) error {
+
+	log.Infof("[Status]: FilePath for App Deployer is %v", appVars.filePath)
+
+	if err := UpdateSockShop("/var/run/"+appVars.filePath, appVars.namespace, appVars.operation); err != nil {
+		return err
+	}
+
+	if err := CheckPodStatusForRevert(appVars.namespace, "app=sock-shop", appVars.timeout, 2, appVars.operation, clientset); err != nil {
+		return err
+	}
+
+	if err := DeleteNamespace(clientset, appVars.namespace); err != nil {
+		log.Info("[Status]: Namespace not found!")
+	}
+
+	return nil
+}
+
 // CheckApplicationStatus checks the status of the AUT
-func CheckApplicationStatus(appNs, appLabel string, timeout, delay int, clientset *kubernetes.Clientset) error {
-	// Checking whether application containers are in ready state
-	log.Info("[Status]: Checking whether application containers are in ready state")
+func CheckApplicationStatus(appNs, appLabel string, timeout, delay int, operation string, clientset *kubernetes.Clientset) error {
+	//  Checking application containers state
+	log.Info("[Status]: Checking application containers state")
 	err := CheckContainerStatus(appNs, appLabel, timeout, delay, clientset)
 	if err != nil {
 		return err
 	}
-	// Checking whether application pods are in running state
-	log.Info("[Status]: Checking whether application pods are in running state")
+	// Checking application pods state
+	log.Info("[Status]: Checking application pods state")
 	err = CheckPodStatus(appNs, appLabel, timeout, delay, clientset)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+//CheckPodStatusForRevert checks the running application terminated or not
+func CheckPodStatusForRevert(appNs, appLabel string, timeout, delay int, operation string, clientset *kubernetes.Clientset) error {
+	err := retry.
+		Times(uint(timeout / delay)).
+		Wait(time.Duration(delay) * time.Second).
+		Try(func(attempt uint) error {
+			podSpec, err := clientset.CoreV1().Pods(appNs).List(metav1.ListOptions{LabelSelector: appLabel})
+			if err != nil {
+				return errors.Errorf("Unable to find the pods with matching labels, err: %v", err)
```

Comment:
not really labels..we are checking if there is no pod at all.

Reply:
We can add a check to see no chanengines exist here. If exists, log and bail out